### DISCUSSION TITLE
User library

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#1336] [Ideas]
_[This might already be possible, but I can't find it in the settings]_
I have some subcircuits that I commonly use in multiple projects. As it is now I either have to copy them into each project or keep them along the component .dig files in /Applications/Digital/lib which causes them to be removed after each clean install of new releases.

Is there a way to have a "user library" folder elsewhere? I'd be great to have a `~/.digital/lib/` that gets added to the search path for subcircuits.  The file `~/.digital.cfg` could also then be put into the ~/.digital/ -folder

**Comment by SirClickALot** [comment]
It might be more useful and more flexible if the USER library could be merged with the SYSTEM one. A further suggestion is that elements in the USER library that have the same ID as those in the SYSTEM library override those in the SYSTEM library - that way, Digital will not complain as it currently does when an element is 'found' in folder as a dependent of another element in that folder.
